devolutions-gateway/src/recording.rs

```diff
@@ -285,6 +311,28 @@ impl RecordingMessageSender {
             .ok()
             .context("couldn't send UpdateRecordingPolicy message")
     }
+
+    pub(crate) fn on_new_chunk_appended(&self, recording_id: Uuid, tx: oneshot::Sender<()>) -> anyhow::Result<()> {
+        let mut lock = self.flush_map.lock();
+        let senders = lock.entry(recording_id);
+        let senders = senders.or_default();
+        senders.push(tx);
+        Ok(())
+    }
```

Comment:
naming: Intent is not clear. I suggest something like `add_new_chunk_listener` or something like that.